test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJstack.java

```diff
@@ -0,0 +1,231 @@
+		StringBuilder buff = new StringBuilder("jstack output:\n"); //$NON-NLS-1$
+		jpsOutput.forEach(s -> {buff.append(s); buff.append("\n");}); //$NON-NLS-1$
+		log(buff.toString());
+		Optional<String> searchResult =  search(testName, jpsOutput);
+		assertTrue(searchResult.isPresent(), "Method name missing"); //$NON-NLS-1$
+	}
+
+	@Test
+	public void testSynchronizers() throws IOException {
+		String myId = TargetManager.getVmId();
+		ArrayList<String> args = new ArrayList<>();
+		args.add(myId);
+		args.add("-l"); //$NON-NLS-1$
+		List<String> jpsOutput = runCommand(args);
+		StringBuilder buff = new StringBuilder("jstack output:\n"); //$NON-NLS-1$
+		jpsOutput.forEach(s -> {buff.append(s); buff.append("\n");}); //$NON-NLS-1$
+		log(buff.toString());
+		Optional<String> searchResult =  search(testName, jpsOutput);
+		assertTrue(searchResult.isPresent(), "Method name missing"); //$NON-NLS-1$
+	}
+
+	@Test
+	public void testProperties() throws IOException {
+		Properties myProps = System.getProperties();
+		final String TEST_VALUE = "test_property_value"; //$NON-NLS-1$
+		final String TEST_PROPERTY = "test.property"; //$NON-NLS-1$
+		myProps.setProperty(TEST_PROPERTY, TEST_VALUE);
+		String myId = TargetManager.getVmId();
+		ArrayList<String> args = new ArrayList<>();
+		args.add(myId);
+		args.add("-p"); //$NON-NLS-1$
+		List<String> jpsOutput = runCommand(args);
+		StringBuilder buff = new StringBuilder("jstack output:\n"); //$NON-NLS-1$
+		jpsOutput.forEach(s -> {buff.append(s); buff.append("\n");}); //$NON-NLS-1$
+		log(buff.toString());
+		Optional<String> searchResult =  search(TEST_VALUE, jpsOutput);
+		assertTrue(searchResult.isPresent(), TEST_PROPERTY + " missing"); //$NON-NLS-1$
+	}
+
+	@BeforeMethod
+	protected void setUp(Method testMethod) {
+		testName = testMethod.getName();
+		log("------------------------------------\nstarting " + testName); //$NON-NLS-1$
+	}
+
+	@BeforeSuite
+	protected void setupSuite() {
+		assertTrue("This test is valid only on OpenJ9",  //$NON-NLS-1$
+				System.getProperty("java.vm.vendor").contains("OpenJ9"));  //$NON-NLS-1$//$NON-NLS-2$
+				final String commandRoot = JSTACK_COMMAND;
+				getJdkUtilityPath(commandRoot);
+	}
+
+	protected void compareThreadInfo(ThreadInfo thrInfo, Thread testThread) {
+		assertEquals(thrInfo.getThreadId(), testThread.getId(), "wrong ID:"); //$NON-NLS-1$
+		assertEquals(thrInfo.getThreadName(), testThread.getName(), "wrong name:"); //$NON-NLS-1$
+		assertEquals(thrInfo.getThreadState(), testThread.getState(), "wrong state:"); //$NON-NLS-1$
+		StackTraceElement[] expectedStackTrace = testThread.getStackTrace();
+		StackTraceElement[] actualStackTrace = testThread.getStackTrace();
+		compareStackTraces(expectedStackTrace, actualStackTrace);
+	}
+
+	protected void compareStackTraces(StackTraceElement[] expectedStackTrace, StackTraceElement[] actualStackTrace) {
+		boolean match = (expectedStackTrace.length == actualStackTrace.length);
+		if (match) {
```

Comment:
This is mostly a reimplementation of `Arrays.equals()` which should be used instead.